diff -Nru v4.0.3/Gemfile v4.1.0/Gemfile
--- v4.0.3/Gemfile	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/Gemfile	2014-04-08 15:48:18.000000000 -0400
@@ -1,45 +1,40 @@
 source 'https://rubygems.org'
 
-# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.0.3'
 
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '4.1.0'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
-
 # Use SCSS for stylesheets
-gem 'sass-rails', '~> 4.0.0'
-
+gem 'sass-rails', '~> 4.0.3'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
-
 # Use CoffeeScript for .js.coffee assets and views
 gem 'coffee-rails', '~> 4.0.0'
-
 # See https://github.com/sstephenson/execjs#readme for more supported runtimes
-# gem 'therubyracer', platforms: :ruby
+# gem 'therubyracer',  platforms: :ruby
 
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
-
 # Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
 gem 'turbolinks'
-
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 1.2'
+gem 'jbuilder', '~> 2.0'
+# bundle exec rake doc:rails generates the API under doc/api.
+gem 'sdoc', '~> 0.4.0',          group: :doc
 
-group :doc do
-  # bundle exec rake doc:rails generates the API under doc/api.
-  gem 'sdoc', require: false
-end
+# Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+gem 'spring',        group: :development
 
 # Use ActiveModel has_secure_password
-# gem 'bcrypt-ruby', '~> 3.1.2'
+# gem 'bcrypt', '~> 3.1.7'
 
 # Use unicorn as the app server
 # gem 'unicorn'
 
 # Use Capistrano for deployment
-# gem 'capistrano', group: :development
+# gem 'capistrano-rails', group: :development
 
 # Use debugger
 # gem 'debugger', group: [:development, :test]
+
diff -Nru v4.0.3/Rakefile v4.1.0/Rakefile
--- v4.0.3/Rakefile	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/Rakefile	2014-04-08 15:48:18.000000000 -0400
@@ -3,4 +3,4 @@
 
 require File.expand_path('../config/application', __FILE__)
 
-Railsdiff::Application.load_tasks
+Rails.application.load_tasks
diff -Nru v4.0.3/app/assets/stylesheets/application.css v4.1.0/app/assets/stylesheets/application.css
--- v4.0.3/app/assets/stylesheets/application.css	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/app/assets/stylesheets/application.css	2014-04-08 15:48:18.000000000 -0400
@@ -5,9 +5,11 @@
  * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
  * or vendor/assets/stylesheets of plugins, if any, can be referenced here using a relative path.
  *
- * You're free to add application-wide styles to this file and they'll appear at the top of the
- * compiled file, but it's generally better to create a new file per style scope.
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any styles
+ * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
+ * file per style scope.
  *
- *= require_self
  *= require_tree .
+ *= require_self
  */
diff -Nru v4.0.3/app/views/layouts/application.html.erb v4.1.0/app/views/layouts/application.html.erb
--- v4.0.3/app/views/layouts/application.html.erb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/app/views/layouts/application.html.erb	2014-04-08 15:48:18.000000000 -0400
@@ -2,8 +2,8 @@
 <html>
 <head>
   <title>Railsdiff</title>
-  <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
-  <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
diff -Nru v4.0.3/config/application.rb v4.1.0/config/application.rb
--- v4.0.3/config/application.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/application.rb	2014-04-08 15:48:18.000000000 -0400
@@ -4,7 +4,7 @@
 
 # Require the gems listed in Gemfile, including any gems
 # you've limited to :test, :development, or :production.
-Bundler.require(:default, Rails.env)
+Bundler.require(*Rails.groups)
 
 module Railsdiff
   class Application < Rails::Application
diff -Nru v4.0.3/config/boot.rb v4.1.0/config/boot.rb
--- v4.0.3/config/boot.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/boot.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,4 +1,4 @@
 # Set up gems listed in the Gemfile.
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
-require 'bundler/setup' if File.exists?(ENV['BUNDLE_GEMFILE'])
+require 'bundler/setup' if File.exist?(ENV['BUNDLE_GEMFILE'])
diff -Nru v4.0.3/config/database.yml v4.1.0/config/database.yml
--- v4.0.3/config/database.yml	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/database.yml	2014-04-08 15:48:18.000000000 -0400
@@ -3,23 +3,23 @@
 #
 #   Ensure the SQLite 3 gem is defined in your Gemfile
 #   gem 'sqlite3'
-development:
+#
+default: &default
   adapter: sqlite3
-  database: db/development.sqlite3
   pool: 5
   timeout: 5000
 
+development:
+  <<: *default
+  database: db/development.sqlite3
+
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
+  <<: *default
   database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
 
 production:
-  adapter: sqlite3
+  <<: *default
   database: db/production.sqlite3
-  pool: 5
-  timeout: 5000
diff -Nru v4.0.3/config/environment.rb v4.1.0/config/environment.rb
--- v4.0.3/config/environment.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/environment.rb	2014-04-08 15:48:18.000000000 -0400
@@ -2,4 +2,4 @@
 require File.expand_path('../application', __FILE__)
 
 # Initialize the Rails application.
-Railsdiff::Application.initialize!
+Rails.application.initialize!
diff -Nru v4.0.3/config/environments/development.rb v4.1.0/config/environments/development.rb
--- v4.0.3/config/environments/development.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/environments/development.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,4 +1,4 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # In the development environment your application's code is reloaded on
@@ -19,11 +19,19 @@
   # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  # Raise an error on page load if there are pending migrations
+  # Raise an error on page load if there are pending migrations.
   config.active_record.migration_error = :page_load
 
   # Debug mode disables concatenation and preprocessing of assets.
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
   config.assets.debug = true
+
+  # Adds additional error checking when serving assets at runtime.
+  # Checks for improperly declared sprockets dependencies.
+  # Raises helpful error messages.
+  config.assets.raise_runtime_errors = true
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
diff -Nru v4.0.3/config/environments/production.rb v4.1.0/config/environments/production.rb
--- v4.0.3/config/environments/production.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/environments/production.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,11 +1,11 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # Code is not reloaded between requests.
   config.cache_classes = true
 
   # Eager load code on boot. This eager loads most of Rails and
-  # your application in memory, allowing both thread web servers
+  # your application in memory, allowing both threaded web servers
   # and those relying on copy on write to perform better.
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
@@ -66,7 +66,7 @@
   # config.action_mailer.raise_delivery_errors = false
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
-  # the I18n.default_locale when a translation can not be found).
+  # the I18n.default_locale when a translation cannot be found).
   config.i18n.fallbacks = true
 
   # Send deprecation notices to registered listeners.
@@ -77,4 +77,7 @@
 
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
 end
diff -Nru v4.0.3/config/environments/test.rb v4.1.0/config/environments/test.rb
--- v4.0.3/config/environments/test.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/environments/test.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,4 +1,4 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # The test environment is used exclusively to run your application's
@@ -14,7 +14,7 @@
 
   # Configure static asset server for tests with Cache-Control for performance.
   config.serve_static_assets  = true
-  config.static_cache_control = "public, max-age=3600"
+  config.static_cache_control = 'public, max-age=3600'
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
@@ -33,4 +33,7 @@
 
   # Print deprecation notices to the stderr.
   config.active_support.deprecation = :stderr
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
diff -Nru v4.0.3/config/initializers/cookies_serializer.rb v4.1.0/config/initializers/cookies_serializer.rb
--- v4.0.3/config/initializers/cookies_serializer.rb	1969-12-31 19:00:00.000000000 -0500
+++ v4.1.0/config/initializers/cookies_serializer.rb	2014-04-08 15:48:18.000000000 -0400
@@ -0,0 +1,3 @@
+# Be sure to restart your server when you modify this file.
+
+Rails.application.config.action_dispatch.cookies_serializer = :json
\ No newline at end of file
diff -Nru v4.0.3/config/initializers/mime_types.rb v4.1.0/config/initializers/mime_types.rb
--- v4.0.3/config/initializers/mime_types.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/initializers/mime_types.rb	2014-04-08 15:48:18.000000000 -0400
@@ -2,4 +2,3 @@
 
 # Add new mime types for use in respond_to blocks:
 # Mime::Type.register "text/richtext", :rtf
-# Mime::Type.register_alias "text/html", :iphone
diff -Nru v4.0.3/config/initializers/secret_token.rb v4.1.0/config/initializers/secret_token.rb
--- v4.0.3/config/initializers/secret_token.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/initializers/secret_token.rb	1969-12-31 19:00:00.000000000 -0500
@@ -1,12 +0,0 @@
-# Be sure to restart your server when you modify this file.
-
-# Your secret key is used for verifying the integrity of signed cookies.
-# If you change this key, all old signed cookies will become invalid!
-
-# Make sure the secret is at least 30 characters and all random,
-# no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
-
-# Make sure your secret_key_base is kept private
-# if you're sharing your code publicly.
-Railsdiff::Application.config.secret_key_base = 'your-secret-token'
diff -Nru v4.0.3/config/initializers/session_store.rb v4.1.0/config/initializers/session_store.rb
--- v4.0.3/config/initializers/session_store.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/initializers/session_store.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,3 +1,3 @@
 # Be sure to restart your server when you modify this file.
 
-Railsdiff::Application.config.session_store :cookie_store, key: '_railsdiff_session'
+Rails.application.config.session_store :cookie_store, key: '_railsdiff_session'
diff -Nru v4.0.3/config/routes.rb v4.1.0/config/routes.rb
--- v4.0.3/config/routes.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/config/routes.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,4 +1,4 @@
-Railsdiff::Application.routes.draw do
+Rails.application.routes.draw do
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
diff -Nru v4.0.3/config/secrets.yml v4.1.0/config/secrets.yml
--- v4.0.3/config/secrets.yml	1969-12-31 19:00:00.000000000 -0500
+++ v4.1.0/config/secrets.yml	2014-04-08 15:48:19.000000000 -0400
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
+
+# Your secret key is used for verifying the integrity of signed cookies.
+# If you change this key, all old signed cookies will become invalid!
+
+# Make sure the secret is at least 30 characters and all random,
+# no regular words or you'll be exposed to dictionary attacks.
+# You can use `rake secret` to generate a secure secret key.
+
+# Make sure the secrets in this file are kept private
+# if you're sharing your code publicly.
+
+development:
+  secret_key_base: your-secret-token
+
+test:
+  secret_key_base: your-secret-token
+
+# Do not keep production secrets in the repository,
+# instead read values from the environment.
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
diff -Nru v4.0.3/public/404.html v4.1.0/public/404.html
--- v4.0.3/public/404.html	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/public/404.html	2014-04-08 15:48:18.000000000 -0400
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>The page you were looking for doesn't exist (404)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,9 +57,11 @@
 <body>
   <!-- This file lives in public/404.html -->
   <div class="dialog">
-    <h1>The page you were looking for doesn't exist.</h1>
-    <p>You may have mistyped the address or the page may have moved.</p>
+    <div>
+      <h1>The page you were looking for doesn't exist.</h1>
+      <p>You may have mistyped the address or the page may have moved.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru v4.0.3/public/422.html v4.1.0/public/422.html
--- v4.0.3/public/422.html	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/public/422.html	2014-04-08 15:48:18.000000000 -0400
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>The change you wanted was rejected (422)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,9 +57,11 @@
 <body>
   <!-- This file lives in public/422.html -->
   <div class="dialog">
-    <h1>The change you wanted was rejected.</h1>
-    <p>Maybe you tried to change something you didn't have access to.</p>
+    <div>
+      <h1>The change you wanted was rejected.</h1>
+      <p>Maybe you tried to change something you didn't have access to.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru v4.0.3/public/500.html v4.1.0/public/500.html
--- v4.0.3/public/500.html	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/public/500.html	2014-04-08 15:48:18.000000000 -0400
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>We're sorry, but something went wrong (500)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,8 +57,10 @@
 <body>
   <!-- This file lives in public/500.html -->
   <div class="dialog">
-    <h1>We're sorry, but something went wrong.</h1>
+    <div>
+      <h1>We're sorry, but something went wrong.</h1>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru v4.0.3/public/robots.txt v4.1.0/public/robots.txt
--- v4.0.3/public/robots.txt	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/public/robots.txt	2014-04-08 15:48:18.000000000 -0400
@@ -1,4 +1,4 @@
-# See http://www.robotstxt.org/wc/norobots.html for documentation on how to use the robots.txt file
+# See http://www.robotstxt.org/robotstxt.html for documentation on how to use the robots.txt file
 #
 # To ban all spiders from the entire site uncomment the next two lines:
 # User-agent: *
diff -Nru v4.0.3/test/test_helper.rb v4.1.0/test/test_helper.rb
--- v4.0.3/test/test_helper.rb	2014-02-21 22:39:50.000000000 -0500
+++ v4.1.0/test/test_helper.rb	2014-04-08 15:48:18.000000000 -0400
@@ -1,10 +1,8 @@
-ENV["RAILS_ENV"] ||= "test"
+ENV['RAILS_ENV'] ||= 'test'
 require File.expand_path('../../config/environment', __FILE__)
 require 'rails/test_help'
 
 class ActiveSupport::TestCase
-  ActiveRecord::Migration.check_pending!
-
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
   #
   # Note: You'll currently still have to declare fixtures explicitly in integration tests
