diff -Nru v3.1.0.rc2/Gemfile v3.1.0.rc7/Gemfile
--- v3.1.0.rc2/Gemfile	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/Gemfile	2012-08-01 09:42:13.000000000 -0400
@@ -1,16 +1,20 @@
 source 'http://rubygems.org'
 
-gem 'rails', '3.1.0.rc2'
+gem 'rails', '3.1.0.rc7'
 
 # Bundle edge Rails instead:
 # gem 'rails',     :git => 'git://github.com/rails/rails.git'
 
 gem 'sqlite3'
 
-# Asset template engines
-gem 'sass-rails', "~> 3.1.0.rc"
-gem 'coffee-script'
-gem 'uglifier'
+
+# Gems used only for assets and not required
+# in production environments by default.
+group :assets do
+  gem 'sass-rails', "  ~> 3.1.0.rc"
+  gem 'coffee-rails', "~> 3.1.0.rc"
+  gem 'uglifier'
+end
 
 gem 'jquery-rails'
 
diff -Nru v3.1.0.rc2/config/application.rb v3.1.0.rc7/config/application.rb
--- v3.1.0.rc2/config/application.rb	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/config/application.rb	2012-08-01 09:42:13.000000000 -0400
@@ -2,9 +2,12 @@
 
 require 'rails/all'
 
-# If you have a Gemfile, require the gems listed there, including any gems
-# you've limited to :test, :development, or :production.
-Bundler.require(:default, Rails.env) if defined?(Bundler)
+if defined?(Bundler)
+  # If you precompile assets before deploying to production, use this line
+  Bundler.require *Rails.groups(:assets => %w(development test))
+  # If you want your assets lazily compiled in production, use this line
+  # Bundler.require(:default, :assets, Rails.env)
+end
 
 module Railsdiff
   class Application < Rails::Application
diff -Nru v3.1.0.rc2/config/environments/development.rb v3.1.0.rc7/config/environments/development.rb
--- v3.1.0.rc2/config/environments/development.rb	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/config/environments/development.rb	2012-08-01 09:42:13.000000000 -0400
@@ -24,4 +24,10 @@
 
   # Do not compress assets
   config.assets.compress = false
+
+  # Allow pass debug_assets=true as a query parameter to load pages with unpackaged assets
+  config.assets.allow_debugging = true
+
+  # Expands the lines which load the assets
+  config.assets.debug = true
 end
diff -Nru v3.1.0.rc2/config/environments/production.rb v3.1.0.rc7/config/environments/production.rb
--- v3.1.0.rc2/config/environments/production.rb	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/config/environments/production.rb	2012-08-01 09:42:13.000000000 -0400
@@ -14,12 +14,9 @@
   # Compress JavaScripts and CSS
   config.assets.compress = true
 
-  # Specify the default JavaScript compressor
-  config.assets.js_compressor  = :uglifier
-
   # Specifies the header that your server uses for sending files
-  # (comment out if your front-end server doesn't support this)
-  config.action_dispatch.x_sendfile_header = "X-Sendfile" # Use 'X-Accel-Redirect' for nginx
+  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
diff -Nru v3.1.0.rc2/config/environments/test.rb v3.1.0.rc7/config/environments/test.rb
--- v3.1.0.rc2/config/environments/test.rb	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/config/environments/test.rb	2012-08-01 09:42:13.000000000 -0400
@@ -36,4 +36,7 @@
 
   # Print deprecation notices to the stderr
   config.active_support.deprecation = :stderr
+
+  # Allow pass debug_assets=true as a query parameter to load pages with unpackaged assets
+  config.assets.allow_debugging = true
 end
diff -Nru v3.1.0.rc2/config/initializers/wrap_parameters.rb v3.1.0.rc7/config/initializers/wrap_parameters.rb
--- v3.1.0.rc2/config/initializers/wrap_parameters.rb	2012-08-01 09:42:03.000000000 -0400
+++ v3.1.0.rc7/config/initializers/wrap_parameters.rb	2012-08-01 09:42:13.000000000 -0400
@@ -4,9 +4,11 @@
 # is enabled by default.
 
 # Enable parameter wrapping for JSON. You can disable this by setting :format to an empty array.
-ActionController::Base.wrap_parameters format: [:json]
+ActiveSupport.on_load(:action_controller) do
+  wrap_parameters format: [:json]
+end
 
 # Disable root element in JSON by default.
-if defined?(ActiveRecord)
-  ActiveRecord::Base.include_root_in_json = false
+ActiveSupport.on_load(:active_record) do
+  self.include_root_in_json = false
 end
