diff -Nru v3.2.7/Gemfile v3.2.14/Gemfile
--- v3.2.7/Gemfile	2012-08-01 23:18:52.000000000 -0400
+++ v3.2.14/Gemfile	2013-07-29 16:16:37.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.7'
+gem 'rails', '3.2.14'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
diff -Nru v3.2.7/README.rdoc v3.2.14/README.rdoc
--- v3.2.7/README.rdoc	2012-08-01 23:18:52.000000000 -0400
+++ v3.2.14/README.rdoc	2013-07-29 16:16:37.000000000 -0400
@@ -157,9 +157,9 @@
 
   |-- app
   |   |-- assets
-  |       |-- images
-  |       |-- javascripts
-  |       `-- stylesheets
+  |   |   |-- images
+  |   |   |-- javascripts
+  |   |   `-- stylesheets
   |   |-- controllers
   |   |-- helpers
   |   |-- mailers
@@ -173,6 +173,7 @@
   |-- db
   |-- doc
   |-- lib
+  |   |-- assets
   |   `-- tasks
   |-- log
   |-- public
@@ -184,13 +185,12 @@
   |   |-- performance
   |   `-- unit
   |-- tmp
-  |   |-- cache
-  |   |-- pids
-  |   |-- sessions
-  |   `-- sockets
+  |   `-- cache
+  |       `-- assets
   `-- vendor
       |-- assets
-          `-- stylesheets
+      |   |-- javascripts
+      |   `-- stylesheets
       `-- plugins
 
 app
