diff -Nru -x '*.png' v4.2.1.rc1/Gemfile v4.2.1.rc2/Gemfile
--- v4.2.1.rc1/Gemfile	2015-02-20 18:51:13.000000000 -0500
+++ v4.2.1.rc2/Gemfile	2015-02-25 17:38:30.000000000 -0500
@@ -2,7 +2,7 @@
 
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.2.1.rc1'
+gem 'rails', '4.2.1.rc2'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
 # Use SCSS for stylesheets
