diff -Nru v4.1.1/Gemfile v4.1.2.rc1/Gemfile
--- v4.1.1/Gemfile	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc1/Gemfile	2014-05-27 09:56:31.000000000 -0700
@@ -2,7 +2,7 @@
 
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.1.1'
+gem 'rails', '4.1.2.rc1'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
 # Use SCSS for stylesheets
diff -Nru v4.1.1/Gemfile.lock v4.1.2.rc1/Gemfile.lock
--- v4.1.1/Gemfile.lock	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.2.rc1/Gemfile.lock	2014-05-27 09:57:43.000000000 -0700
@@ -0,0 +1,125 @@
+GEM
+  remote: https://rubygems.org/
+  specs:
+    actionmailer (4.1.2.rc1)
+      actionpack (= 4.1.2.rc1)
+      actionview (= 4.1.2.rc1)
+      mail (~> 2.5.4)
+    actionpack (4.1.2.rc1)
+      actionview (= 4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      rack (~> 1.5.2)
+      rack-test (~> 0.6.2)
+    actionview (4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      builder (~> 3.1)
+      erubis (~> 2.7.0)
+    activemodel (4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      builder (~> 3.1)
+    activerecord (4.1.2.rc1)
+      activemodel (= 4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      arel (~> 5.0.0)
+    activesupport (4.1.2.rc1)
+      i18n (~> 0.6, >= 0.6.9)
+      json (~> 1.7, >= 1.7.7)
+      minitest (~> 5.1)
+      thread_safe (~> 0.1)
+      tzinfo (~> 1.1)
+    arel (5.0.1.20140414130214)
+    builder (3.2.2)
+    coffee-rails (4.0.1)
+      coffee-script (>= 2.2.0)
+      railties (>= 4.0.0, < 5.0)
+    coffee-script (2.2.0)
+      coffee-script-source
+      execjs
+    coffee-script-source (1.7.0)
+    erubis (2.7.0)
+    execjs (2.0.2)
+    hike (1.2.3)
+    i18n (0.6.9)
+    jbuilder (2.0.7)
+      activesupport (>= 3.0.0, < 5)
+      multi_json (~> 1.2)
+    jquery-rails (3.1.0)
+      railties (>= 3.0, < 5.0)
+      thor (>= 0.14, < 2.0)
+    json (1.8.1)
+    mail (2.5.4)
+      mime-types (~> 1.16)
+      treetop (~> 1.4.8)
+    mime-types (1.25.1)
+    minitest (5.3.4)
+    multi_json (1.10.1)
+    polyglot (0.3.4)
+    rack (1.5.2)
+    rack-test (0.6.2)
+      rack (>= 1.0)
+    rails (4.1.2.rc1)
+      actionmailer (= 4.1.2.rc1)
+      actionpack (= 4.1.2.rc1)
+      actionview (= 4.1.2.rc1)
+      activemodel (= 4.1.2.rc1)
+      activerecord (= 4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      bundler (>= 1.3.0, < 2.0)
+      railties (= 4.1.2.rc1)
+      sprockets-rails (~> 2.0)
+    railties (4.1.2.rc1)
+      actionpack (= 4.1.2.rc1)
+      activesupport (= 4.1.2.rc1)
+      rake (>= 0.8.7)
+      thor (>= 0.18.1, < 2.0)
+    rake (10.3.2)
+    rdoc (4.1.1)
+      json (~> 1.4)
+    sass (3.2.19)
+    sass-rails (4.0.3)
+      railties (>= 4.0.0, < 5.0)
+      sass (~> 3.2.0)
+      sprockets (~> 2.8, <= 2.11.0)
+      sprockets-rails (~> 2.0)
+    sdoc (0.4.0)
+      json (~> 1.8)
+      rdoc (~> 4.0, < 5.0)
+    spring (1.1.3)
+    sprockets (2.11.0)
+      hike (~> 1.2)
+      multi_json (~> 1.0)
+      rack (~> 1.0)
+      tilt (~> 1.1, != 1.3.0)
+    sprockets-rails (2.1.3)
+      actionpack (>= 3.0)
+      activesupport (>= 3.0)
+      sprockets (~> 2.8)
+    sqlite3 (1.3.9)
+    thor (0.19.1)
+    thread_safe (0.3.3)
+    tilt (1.4.1)
+    treetop (1.4.15)
+      polyglot
+      polyglot (>= 0.3.1)
+    turbolinks (2.2.2)
+      coffee-rails
+    tzinfo (1.2.0)
+      thread_safe (~> 0.1)
+    uglifier (2.5.0)
+      execjs (>= 0.3.0)
+      json (>= 1.8.0)
+
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  coffee-rails (~> 4.0.0)
+  jbuilder (~> 2.0)
+  jquery-rails
+  rails (= 4.1.2.rc1)
+  sass-rails (~> 4.0.3)
+  sdoc (~> 0.4.0)
+  spring
+  sqlite3
+  turbolinks
+  uglifier (>= 1.3.0)
diff -Nru v4.1.1/bin/rails v4.1.2.rc1/bin/rails
--- v4.1.1/bin/rails	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc1/bin/rails	2014-05-27 09:57:46.000000000 -0700
@@ -1,4 +1,8 @@
 #!/usr/bin/env ruby
+begin
+  load File.expand_path("../spring", __FILE__)
+rescue LoadError
+end
 APP_PATH = File.expand_path('../../config/application',  __FILE__)
 require_relative '../config/boot'
 require 'rails/commands'
diff -Nru v4.1.1/bin/rake v4.1.2.rc1/bin/rake
--- v4.1.1/bin/rake	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc1/bin/rake	2014-05-27 09:57:46.000000000 -0700
@@ -1,4 +1,8 @@
 #!/usr/bin/env ruby
+begin
+  load File.expand_path("../spring", __FILE__)
+rescue LoadError
+end
 require_relative '../config/boot'
 require 'rake'
 Rake.application.run
diff -Nru v4.1.1/bin/spring v4.1.2.rc1/bin/spring
--- v4.1.1/bin/spring	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.2.rc1/bin/spring	2014-05-27 09:57:46.000000000 -0700
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
+
+# This file loads spring without using Bundler, in order to be fast
+# It gets overwritten when you run the `spring binstub` command
+
+unless defined?(Spring)
+  require "rubygems"
+  require "bundler"
+
+  if match = Bundler.default_lockfile.read.match(/^GEM$.*?^    spring \((.*?)\)$.*?^$/m)
+    ENV["GEM_PATH"] = ([Bundler.bundle_path.to_s] + Gem.path).join(File::PATH_SEPARATOR)
+    ENV["GEM_HOME"] = ""
+    Gem.paths = ENV
+
+    gem "spring", match[1]
+    require "spring/binstub"
+  end
+end
diff -Nru v4.1.1/config/environments/production.rb v4.1.2.rc1/config/environments/production.rb
--- v4.1.1/config/environments/production.rb	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc1/config/environments/production.rb	2014-05-27 09:56:31.000000000 -0700
@@ -32,8 +32,7 @@
   # Generate digests for assets URLs.
   config.assets.digest = true
 
-  # Version of your assets, change this if you want to expire all your assets.
-  config.assets.version = '1.0'
+  # `config.assets.precompile` has moved to config/initializers/assets.rb
 
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
diff -Nru v4.1.1/config/initializers/assets.rb v4.1.2.rc1/config/initializers/assets.rb
--- v4.1.1/config/initializers/assets.rb	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.2.rc1/config/initializers/assets.rb	2014-05-27 09:56:31.000000000 -0700
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+
+# Precompile additional assets.
+# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+# Rails.application.config.assets.precompile += %w( search.js )
diff -Nru v4.1.1/test/test_helper.rb v4.1.2.rc1/test/test_helper.rb
--- v4.1.1/test/test_helper.rb	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc1/test/test_helper.rb	2014-05-27 09:56:31.000000000 -0700
@@ -4,9 +4,6 @@
 
 class ActiveSupport::TestCase
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
-  #
-  # Note: You'll currently still have to declare fixtures explicitly in integration tests
-  # -- they do not yet inherit this setting
   fixtures :all
 
   # Add more helper methods to be used by all tests here...
