diff --git a/Gemfile b/Gemfile
index 45ffe90..0ace91f 100644
--- a/Gemfile
+++ b/Gemfile
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.5'
+gem 'rails', '3.2.6'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
diff --git a/config/initializers/secret_token.rb b/config/initializers/secret_token.rb
index 576ff9a..b7a63f3 100644
--- a/config/initializers/secret_token.rb
+++ b/config/initializers/secret_token.rb
@@ -4,4 +4,4 @@
 # If you change this key, all old signed cookies will become invalid!
 # Make sure the secret is at least 30 characters and all random,
 # no regular words or you'll be exposed to dictionary attacks.
-App::Application.config.secret_token = '4b9b9e2d40897509cfdfea227dbe0d53111b3605ead238af9af6062db1b5848b46a6611d369c16c6521cb85e9f6dc948829757b48b005939a6062186d369b705'
+App::Application.config.secret_token = '62aae892746c9ad86c9dacd7da144aa33d6da2e596693b3eacd446335ae50bdeba3c7a8d529c090da70ae789cec7baa81b8ae577f00a9fbe6576953102ccfae1'
