diff -Nru v3.1.0.rc3/Gemfile v3.1.1/Gemfile
--- v3.1.0.rc3/Gemfile	2012-08-01 09:42:05.000000000 -0400
+++ v3.1.1/Gemfile	2012-07-26 23:32:35.000000000 -0400
@@ -1,19 +1,26 @@
 source 'http://rubygems.org'
 
-gem 'rails', '3.1.0.rc3'
+gem 'rails', '3.1.1'
 
 # Bundle edge Rails instead:
 # gem 'rails',     :git => 'git://github.com/rails/rails.git'
 
 gem 'sqlite3'
 
-# Asset template engines
-gem 'sass-rails', "~> 3.1.0.rc"
-gem 'coffee-script'
-gem 'uglifier'
+
+# Gems used only for assets and not required
+# in production environments by default.
+group :assets do
+  gem 'sass-rails',   '~> 3.1.4'
+  gem 'coffee-rails', '~> 3.1.1'
+  gem 'uglifier', '>= 1.0.3'
+end
 
 gem 'jquery-rails'
 
+# To use ActiveModel has_secure_password
+# gem 'bcrypt-ruby', '~> 3.0.0'
+
 # Use unicorn as the web server
 # gem 'unicorn'
 
diff -Nru v3.1.0.rc3/config/application.rb v3.1.1/config/application.rb
--- v3.1.0.rc3/config/application.rb	2012-08-01 09:42:05.000000000 -0400
+++ v3.1.1/config/application.rb	2012-07-26 23:32:35.000000000 -0400
@@ -2,9 +2,12 @@
 
 require 'rails/all'
 
-# If you have a Gemfile, require the gems listed there, including any gems
-# you've limited to :test, :development, or :production.
-Bundler.require(:default, Rails.env) if defined?(Bundler)
+if defined?(Bundler)
+  # If you precompile assets before deploying to production, use this line
+  Bundler.require(*Rails.groups(:assets => %w(development test)))
+  # If you want your assets lazily compiled in production, use this line
+  # Bundler.require(:default, :assets, Rails.env)
+end
 
 module Railsdiff
   class Application < Rails::Application
@@ -38,5 +41,8 @@
 
     # Enable the asset pipeline
     config.assets.enabled = true
+
+    # Version of your assets, change this if you want to expire all your assets
+    config.assets.version = '1.0'
   end
 end
diff -Nru v3.1.0.rc3/config/environments/development.rb v3.1.1/config/environments/development.rb
--- v3.1.0.rc3/config/environments/development.rb	2012-08-01 09:42:05.000000000 -0400
+++ v3.1.1/config/environments/development.rb	2012-07-26 23:32:35.000000000 -0400
@@ -24,4 +24,7 @@
 
   # Do not compress assets
   config.assets.compress = false
+
+  # Expands the lines which load the assets
+  config.assets.debug = true
 end
diff -Nru v3.1.0.rc3/config/environments/production.rb v3.1.1/config/environments/production.rb
--- v3.1.0.rc3/config/environments/production.rb	2012-08-01 09:42:05.000000000 -0400
+++ v3.1.1/config/environments/production.rb	2012-07-26 23:32:35.000000000 -0400
@@ -14,12 +14,18 @@
   # Compress JavaScripts and CSS
   config.assets.compress = true
 
-  # Specify the default JavaScript compressor
-  config.assets.js_compressor  = :uglifier
+  # Don't fallback to assets pipeline if a precompiled asset is missed
+  config.assets.compile = false
+
+  # Generate digests for assets URLs
+  config.assets.digest = true
+
+  # Defaults to Rails.root.join("public/assets")
+  # config.assets.manifest = YOUR_PATH
 
   # Specifies the header that your server uses for sending files
-  # (comment out if your front-end server doesn't support this)
-  config.action_dispatch.x_sendfile_header = "X-Sendfile" # Use 'X-Accel-Redirect' for nginx
+  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
diff -Nru v3.1.0.rc3/config/initializers/wrap_parameters.rb v3.1.1/config/initializers/wrap_parameters.rb
--- v3.1.0.rc3/config/initializers/wrap_parameters.rb	2012-08-01 09:42:05.000000000 -0400
+++ v3.1.1/config/initializers/wrap_parameters.rb	2012-07-26 23:32:35.000000000 -0400
@@ -4,9 +4,11 @@
 # is enabled by default.
 
 # Enable parameter wrapping for JSON. You can disable this by setting :format to an empty array.
-ActionController::Base.wrap_parameters format: [:json]
+ActiveSupport.on_load(:action_controller) do
+  wrap_parameters format: [:json]
+end
 
 # Disable root element in JSON by default.
-if defined?(ActiveRecord)
-  ActiveRecord::Base.include_root_in_json = false
+ActiveSupport.on_load(:active_record) do
+  self.include_root_in_json = false
 end
