diff -Nru v3.1.1.rc2/.gitignore v3.1.1/.gitignore
--- v3.1.1.rc2/.gitignore	1969-12-31 19:00:00.000000000 -0500
+++ v3.1.1/.gitignore	2012-07-26 23:32:35.000000000 -0400
@@ -0,0 +1,5 @@
+.bundle
+db/*.sqlite3
+log/*.log
+tmp/
+.sass-cache/
diff -Nru v3.1.1.rc2/Gemfile v3.1.1/Gemfile
--- v3.1.1.rc2/Gemfile	2012-07-26 23:32:02.000000000 -0400
+++ v3.1.1/Gemfile	2012-07-26 23:32:35.000000000 -0400
@@ -1,6 +1,6 @@
 source 'http://rubygems.org'
 
-gem 'rails', '3.1.1.rc2'
+gem 'rails', '3.1.1'
 
 # Bundle edge Rails instead:
 # gem 'rails',     :git => 'git://github.com/rails/rails.git'
@@ -11,8 +11,8 @@
 # Gems used only for assets and not required
 # in production environments by default.
 group :assets do
-  gem 'sass-rails',   '~> 3.1.0'
-  gem 'coffee-rails', '~> 3.1.0'
+  gem 'sass-rails',   '~> 3.1.4'
+  gem 'coffee-rails', '~> 3.1.1'
   gem 'uglifier', '>= 1.0.3'
 end
 
diff -Nru v3.1.1.rc2/railsdiff/.gitignore v3.1.1/railsdiff/.gitignore
--- v3.1.1.rc2/railsdiff/.gitignore	2012-07-26 23:32:02.000000000 -0400
+++ v3.1.1/railsdiff/.gitignore	1969-12-31 19:00:00.000000000 -0500
@@ -1,5 +0,0 @@
-.bundle
-db/*.sqlite3
-log/*.log
-tmp/
-.sass-cache/
