diff -ur tmp/generated/v3.2.0.rc2/railsdiff/Gemfile tmp/generated/v3.2.3.rc1/railsdiff/Gemfile
--- tmp/generated/v3.2.0.rc2/railsdiff/Gemfile	2012-07-26 16:03:26.000000000 -0400
+++ tmp/generated/v3.2.3.rc1/railsdiff/Gemfile	2012-07-26 16:03:10.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.0.rc2'
+gem 'rails', '3.2.3.rc1'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
@@ -12,7 +12,10 @@
 # in production environments by default.
 group :assets do
   gem 'sass-rails',   '~> 3.2.3'
-  gem 'coffee-rails', '~> 3.2.0'
+  gem 'coffee-rails', '~> 3.2.1'
+
+  # See https://github.com/sstephenson/execjs#readme for more supported runtimes
+  # gem 'therubyracer', :platform => :ruby
 
   gem 'uglifier', '>= 1.0.3'
 end
@@ -25,7 +28,7 @@
 # To use Jbuilder templates for JSON
 # gem 'jbuilder'
 
-# Use unicorn as the web server
+# Use unicorn as the app server
 # gem 'unicorn'
 
 # Deploy with Capistrano
diff -ur tmp/generated/v3.2.0.rc2/railsdiff/config/application.rb tmp/generated/v3.2.3.rc1/railsdiff/config/application.rb
--- tmp/generated/v3.2.0.rc2/railsdiff/config/application.rb	2012-07-26 16:03:26.000000000 -0400
+++ tmp/generated/v3.2.3.rc1/railsdiff/config/application.rb	2012-07-26 16:03:10.000000000 -0400
@@ -48,7 +48,7 @@
     # This will create an empty whitelist of attributes available for mass-assignment for all models
     # in your app. As such, your models will need to explicitly whitelist or blacklist accessible
     # parameters by using an attr_accessible or attr_protected declaration.
-    # config.active_record.whitelist_attributes = true
+    config.active_record.whitelist_attributes = true
 
     # Enable the asset pipeline
     config.assets.enabled = true
diff -ur tmp/generated/v3.2.0.rc2/railsdiff/config/environments/development.rb tmp/generated/v3.2.3.rc1/railsdiff/config/environments/development.rb
--- tmp/generated/v3.2.0.rc2/railsdiff/config/environments/development.rb	2012-07-26 16:03:26.000000000 -0400
+++ tmp/generated/v3.2.3.rc1/railsdiff/config/environments/development.rb	2012-07-26 16:03:10.000000000 -0400
@@ -22,7 +22,7 @@
   # Only use best-standards-support built into browsers
   config.action_dispatch.best_standards_support = :builtin
 
-  # Raise exception on mass assignment protection for ActiveRecord models
+  # Raise exception on mass assignment protection for Active Record models
   config.active_record.mass_assignment_sanitizer = :strict
 
   # Log the query plan for queries taking more than this (works
diff -ur tmp/generated/v3.2.0.rc2/railsdiff/config/environments/test.rb tmp/generated/v3.2.3.rc1/railsdiff/config/environments/test.rb
--- tmp/generated/v3.2.0.rc2/railsdiff/config/environments/test.rb	2012-07-26 16:03:26.000000000 -0400
+++ tmp/generated/v3.2.3.rc1/railsdiff/config/environments/test.rb	2012-07-26 16:03:10.000000000 -0400
@@ -29,7 +29,7 @@
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
 
-  # Raise exception on mass assignment protection for ActiveRecord models
+  # Raise exception on mass assignment protection for Active Record models
   config.active_record.mass_assignment_sanitizer = :strict
 
   # Print deprecation notices to the stderr
