diff -Nru v3.2.4.rc1/Gemfile v3.2.7.rc1/Gemfile
--- v3.2.4.rc1/Gemfile	2012-07-26 23:33:20.000000000 -0400
+++ v3.2.7.rc1/Gemfile	2012-07-26 23:37:53.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.4.rc1'
+gem 'rails', '3.2.7.rc1'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
@@ -15,7 +15,7 @@
   gem 'coffee-rails', '~> 3.2.1'
 
   # See https://github.com/sstephenson/execjs#readme for more supported runtimes
-  # gem 'therubyracer', :platform => :ruby
+  # gem 'therubyracer', :platforms => :ruby
 
   gem 'uglifier', '>= 1.0.3'
 end
