diff -ur tmp/generated/v3.2.2/railsdiff/Gemfile tmp/generated/v3.2.2.rc1/railsdiff/Gemfile
--- tmp/generated/v3.2.2/railsdiff/Gemfile	2012-07-26 10:43:56.000000000 -0400
+++ tmp/generated/v3.2.2.rc1/railsdiff/Gemfile	2012-07-26 10:43:58.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.2'
+gem 'rails', '3.2.2.rc1'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
