diff -ur tmp/generated/v3.2.0/railsdiff/Gemfile tmp/generated/v3.2.1/railsdiff/Gemfile
--- tmp/generated/v3.2.0/railsdiff/Gemfile	2012-07-26 10:43:47.000000000 -0400
+++ tmp/generated/v3.2.1/railsdiff/Gemfile	2012-07-26 10:43:54.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.0'
+gem 'rails', '3.2.1'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
