diff -ur tmp/generated/v3.2.2/railsdiff/Gemfile tmp/generated/v3.2.4.rc1/railsdiff/Gemfile
--- tmp/generated/v3.2.2/railsdiff/Gemfile	2012-07-26 16:03:19.000000000 -0400
+++ tmp/generated/v3.2.4.rc1/railsdiff/Gemfile	2012-07-26 16:03:06.000000000 -0400
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'rails', '3.2.2'
+gem 'rails', '3.2.4.rc1'
 
 # Bundle edge Rails instead:
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
@@ -15,7 +15,7 @@
   gem 'coffee-rails', '~> 3.2.1'
 
   # See https://github.com/sstephenson/execjs#readme for more supported runtimes
-  # gem 'therubyracer'
+  # gem 'therubyracer', :platform => :ruby
 
   gem 'uglifier', '>= 1.0.3'
 end
@@ -35,4 +35,4 @@
 # gem 'capistrano'
 
 # To use debugger
-# gem 'ruby-debug19', :require => 'ruby-debug'
+# gem 'debugger'
diff -ur tmp/generated/v3.2.2/railsdiff/app/assets/stylesheets/application.css tmp/generated/v3.2.4.rc1/railsdiff/app/assets/stylesheets/application.css
--- tmp/generated/v3.2.2/railsdiff/app/assets/stylesheets/application.css	2012-07-26 16:03:19.000000000 -0400
+++ tmp/generated/v3.2.4.rc1/railsdiff/app/assets/stylesheets/application.css	2012-07-26 16:03:06.000000000 -0400
@@ -10,4 +10,4 @@
  *
  *= require_self
  *= require_tree .
-*/
+ */
diff -ur tmp/generated/v3.2.2/railsdiff/config/application.rb tmp/generated/v3.2.4.rc1/railsdiff/config/application.rb
--- tmp/generated/v3.2.2/railsdiff/config/application.rb	2012-07-26 16:03:19.000000000 -0400
+++ tmp/generated/v3.2.4.rc1/railsdiff/config/application.rb	2012-07-26 16:03:06.000000000 -0400
@@ -39,6 +39,9 @@
     # Configure sensitive parameters which will be filtered from the log file.
     config.filter_parameters += [:password]
 
+    # Enable escaping HTML in JSON.
+    config.active_support.escape_html_entities_in_json = true
+
     # Use SQL instead of Active Record's schema dumper when creating the database.
     # This is necessary if your schema can't be completely dumped by the schema dumper,
     # like if you have constraints or database-specific column types
@@ -48,7 +51,7 @@
     # This will create an empty whitelist of attributes available for mass-assignment for all models
     # in your app. As such, your models will need to explicitly whitelist or blacklist accessible
     # parameters by using an attr_accessible or attr_protected declaration.
-    # config.active_record.whitelist_attributes = true
+    config.active_record.whitelist_attributes = true
 
     # Enable the asset pipeline
     config.assets.enabled = true
diff -ur tmp/generated/v3.2.2/railsdiff/config/environments/production.rb tmp/generated/v3.2.4.rc1/railsdiff/config/environments/production.rb
--- tmp/generated/v3.2.2/railsdiff/config/environments/production.rb	2012-07-26 16:03:19.000000000 -0400
+++ tmp/generated/v3.2.4.rc1/railsdiff/config/environments/production.rb	2012-07-26 16:03:06.000000000 -0400
@@ -20,7 +20,7 @@
   # Generate digests for assets URLs
   config.assets.digest = true
 
-  # Defaults to Rails.root.join("public/assets")
+  # Defaults to nil and saved in location specified by config.assets.prefix
   # config.assets.manifest = YOUR_PATH
 
   # Specifies the header that your server uses for sending files
