diff -Nru -x '*.png' v4.0.5/Gemfile v4.0.10/Gemfile
--- v4.0.5/Gemfile	2014-05-06 14:06:47.000000000 -0400
+++ v4.0.10/Gemfile	2014-09-11 14:00:03.000000000 -0400
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.0.5'
+gem 'rails', '4.0.10'
 
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
diff -Nru -x '*.png' v4.0.5/config/environments/development.rb v4.0.10/config/environments/development.rb
--- v4.0.5/config/environments/development.rb	2014-05-06 14:06:47.000000000 -0400
+++ v4.0.10/config/environments/development.rb	2014-09-11 14:00:03.000000000 -0400
@@ -26,4 +26,7 @@
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
   config.assets.debug = true
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
diff -Nru -x '*.png' v4.0.5/config/environments/test.rb v4.0.10/config/environments/test.rb
--- v4.0.5/config/environments/test.rb	2014-05-06 14:06:47.000000000 -0400
+++ v4.0.10/config/environments/test.rb	2014-09-11 14:00:03.000000000 -0400
@@ -33,4 +33,7 @@
 
   # Print deprecation notices to the stderr.
   config.active_support.deprecation = :stderr
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
