diff -Nru -x '*.png' v4.0.1/.gitignore v4.2.0.rc1/.gitignore
--- v4.0.1/.gitignore	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/.gitignore	2014-11-28 23:40:41.000000000 -0500
@@ -12,5 +12,6 @@
 /db/*.sqlite3-journal
 
 # Ignore all logfiles and tempfiles.
-/log/*.log
+/log/*
+!/log/.keep
 /tmp
diff -Nru -x '*.png' v4.0.1/Gemfile v4.2.0.rc1/Gemfile
--- v4.0.1/Gemfile	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/Gemfile	2014-11-28 23:40:41.000000000 -0500
@@ -1,45 +1,45 @@
 source 'https://rubygems.org'
 
-# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.0.1'
 
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '4.2.0.rc1'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
-
 # Use SCSS for stylesheets
-gem 'sass-rails', '~> 4.0.0'
-
+gem 'sass-rails', '~> 4.0'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
-
-# Use CoffeeScript for .js.coffee assets and views
-gem 'coffee-rails', '~> 4.0.0'
-
+# Use CoffeeScript for .coffee assets and views
+gem 'coffee-rails', '~> 4.1.0'
 # See https://github.com/sstephenson/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
 
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
-
 # Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
 gem 'turbolinks'
-
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 1.2'
-
-group :doc do
-  # bundle exec rake doc:rails generates the API under doc/api.
-  gem 'sdoc', require: false
-end
+gem 'jbuilder', '~> 2.0'
+# bundle exec rake doc:rails generates the API under doc/api.
+gem 'sdoc', '~> 0.4.0', group: :doc
 
 # Use ActiveModel has_secure_password
-# gem 'bcrypt-ruby', '~> 3.1.2'
+# gem 'bcrypt', '~> 3.1.7'
 
-# Use unicorn as the app server
+# Use Unicorn as the app server
 # gem 'unicorn'
 
 # Use Capistrano for deployment
-# gem 'capistrano', group: :development
+# gem 'capistrano-rails', group: :development
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug'
+
+  # Access an IRB console on exception pages or by using <%= console %> in views
+  gem 'web-console', '~> 2.0'
+
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem 'spring'
+end
 
-# Use debugger
-# gem 'debugger', group: [:development, :test]
diff -Nru -x '*.png' v4.0.1/Rakefile v4.2.0.rc1/Rakefile
--- v4.0.1/Rakefile	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/Rakefile	2014-11-28 23:40:41.000000000 -0500
@@ -3,4 +3,4 @@
 
 require File.expand_path('../config/application', __FILE__)
 
-Railsdiff::Application.load_tasks
+Rails.application.load_tasks
diff -Nru -x '*.png' v4.0.1/app/assets/javascripts/application.js v4.2.0.rc1/app/assets/javascripts/application.js
--- v4.0.1/app/assets/javascripts/application.js	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/app/assets/javascripts/application.js	2014-11-28 23:40:41.000000000 -0500
@@ -2,7 +2,7 @@
 // listed below.
 //
 // Any JavaScript/Coffee file within this directory, lib/assets/javascripts, vendor/assets/javascripts,
-// or vendor/assets/javascripts of plugins, if any, can be referenced here using a relative path.
+// or any plugin's vendor/assets/javascripts directory can be referenced here using a relative path.
 //
 // It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
 // compiled file.
diff -Nru -x '*.png' v4.0.1/app/assets/stylesheets/application.css v4.2.0.rc1/app/assets/stylesheets/application.css
--- v4.0.1/app/assets/stylesheets/application.css	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/app/assets/stylesheets/application.css	2014-11-28 23:40:41.000000000 -0500
@@ -3,11 +3,13 @@
  * listed below.
  *
  * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
- * or vendor/assets/stylesheets of plugins, if any, can be referenced here using a relative path.
+ * or any plugin's vendor/assets/stylesheets directory can be referenced here using a relative path.
  *
- * You're free to add application-wide styles to this file and they'll appear at the top of the
- * compiled file, but it's generally better to create a new file per style scope.
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any styles
+ * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
+ * file per style scope.
  *
- *= require_self
  *= require_tree .
+ *= require_self
  */
diff -Nru -x '*.png' v4.0.1/app/views/layouts/application.html.erb v4.2.0.rc1/app/views/layouts/application.html.erb
--- v4.0.1/app/views/layouts/application.html.erb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/app/views/layouts/application.html.erb	2014-11-28 23:40:41.000000000 -0500
@@ -2,8 +2,8 @@
 <html>
 <head>
   <title>Railsdiff</title>
-  <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
-  <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
diff -Nru -x '*.png' v4.0.1/bin/rails v4.2.0.rc1/bin/rails
--- v4.0.1/bin/rails	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/bin/rails	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
 #!/usr/bin/env ruby
-APP_PATH = File.expand_path('../../config/application',  __FILE__)
+APP_PATH = File.expand_path('../../config/application', __FILE__)
 require_relative '../config/boot'
 require 'rails/commands'
diff -Nru -x '*.png' v4.0.1/bin/setup v4.2.0.rc1/bin/setup
--- v4.0.1/bin/setup	1969-12-31 19:00:00.000000000 -0500
+++ v4.2.0.rc1/bin/setup	2014-11-28 23:40:41.000000000 -0500
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+require 'pathname'
+
+# path to your application root.
+APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
+
+Dir.chdir APP_ROOT do
+  # This script is a starting point to setup your application.
+  # Add necessary setup steps to this file:
+
+  puts "== Installing dependencies =="
+  system "gem install bundler --conservative"
+  system "bundle check || bundle install"
+
+  # puts "\n== Copying sample files =="
+  # unless File.exist?("config/database.yml")
+  #   system "cp config/database.yml.sample config/database.yml"
+  # end
+
+  puts "\n== Preparing database =="
+  system "bin/rake db:setup"
+
+  puts "\n== Removing old logs and tempfiles =="
+  system "rm -f log/*"
+  system "rm -rf tmp/cache"
+
+  puts "\n== Restarting application server =="
+  system "touch tmp/restart.txt"
+end
diff -Nru -x '*.png' v4.0.1/config/application.rb v4.2.0.rc1/config/application.rb
--- v4.0.1/config/application.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/application.rb	2014-11-28 23:40:41.000000000 -0500
@@ -4,7 +4,7 @@
 
 # Require the gems listed in Gemfile, including any gems
 # you've limited to :test, :development, or :production.
-Bundler.require(:default, Rails.env)
+Bundler.require(*Rails.groups)
 
 module Railsdiff
   class Application < Rails::Application
@@ -19,5 +19,8 @@
     # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
     # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
     # config.i18n.default_locale = :de
+
+    # Do not swallow errors in after_commit/after_rollback callbacks.
+    config.active_record.raise_in_transactional_callbacks = true
   end
 end
diff -Nru -x '*.png' v4.0.1/config/boot.rb v4.2.0.rc1/config/boot.rb
--- v4.0.1/config/boot.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/boot.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,3 @@
-# Set up gems listed in the Gemfile.
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
-require 'bundler/setup' if File.exists?(ENV['BUNDLE_GEMFILE'])
+require 'bundler/setup' # Set up gems listed in the Gemfile.
diff -Nru -x '*.png' v4.0.1/config/database.yml v4.2.0.rc1/config/database.yml
--- v4.0.1/config/database.yml	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/database.yml	2014-11-28 23:40:41.000000000 -0500
@@ -3,23 +3,23 @@
 #
 #   Ensure the SQLite 3 gem is defined in your Gemfile
 #   gem 'sqlite3'
-development:
+#
+default: &default
   adapter: sqlite3
-  database: db/development.sqlite3
   pool: 5
   timeout: 5000
 
+development:
+  <<: *default
+  database: db/development.sqlite3
+
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
+  <<: *default
   database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
 
 production:
-  adapter: sqlite3
+  <<: *default
   database: db/production.sqlite3
-  pool: 5
-  timeout: 5000
diff -Nru -x '*.png' v4.0.1/config/environment.rb v4.2.0.rc1/config/environment.rb
--- v4.0.1/config/environment.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/environment.rb	2014-11-28 23:40:41.000000000 -0500
@@ -2,4 +2,4 @@
 require File.expand_path('../application', __FILE__)
 
 # Initialize the Rails application.
-Railsdiff::Application.initialize!
+Rails.application.initialize!
diff -Nru -x '*.png' v4.0.1/config/environments/development.rb v4.2.0.rc1/config/environments/development.rb
--- v4.0.1/config/environments/development.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/environments/development.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # In the development environment your application's code is reloaded on
@@ -19,11 +19,23 @@
   # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  # Raise an error on page load if there are pending migrations
+  # Raise an error on page load if there are pending migrations.
   config.active_record.migration_error = :page_load
 
   # Debug mode disables concatenation and preprocessing of assets.
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
   config.assets.debug = true
+
+  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
+  # yet still be able to expire them through the digest params.
+  config.assets.digest = true
+
+  # Adds additional error checking when serving assets at runtime.
+  # Checks for improperly declared sprockets dependencies.
+  # Raises helpful error messages.
+  config.assets.raise_runtime_errors = true
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
diff -Nru -x '*.png' v4.0.1/config/environments/production.rb v4.2.0.rc1/config/environments/production.rb
--- v4.0.1/config/environments/production.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/environments/production.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,11 +1,11 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # Code is not reloaded between requests.
   config.cache_classes = true
 
   # Eager load code on boot. This eager loads most of Rails and
-  # your application in memory, allowing both thread web servers
+  # your application in memory, allowing both threaded web servers
   # and those relying on copy on write to perform better.
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
@@ -16,10 +16,11 @@
 
   # Enable Rack::Cache to put a simple HTTP cache in front of your application
   # Add `rack-cache` to your Gemfile before enabling this.
-  # For large-scale production use, consider using a caching reverse proxy like nginx, varnish or squid.
+  # For large-scale production use, consider using a caching reverse proxy like
+  # NGINX, varnish or squid.
   # config.action_dispatch.rack_cache = true
 
-  # Disable Rails's static asset server (Apache or nginx will already do this).
+  # Disable Rails's static asset server (Apache or NGINX will already do this).
   config.serve_static_assets = false
 
   # Compress JavaScripts and CSS.
@@ -29,21 +30,22 @@
   # Do not fallback to assets pipeline if a precompiled asset is missed.
   config.assets.compile = false
 
-  # Generate digests for assets URLs.
+  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
+  # yet still be able to expire them through the digest params.
   config.assets.digest = true
 
-  # Version of your assets, change this if you want to expire all your assets.
-  config.assets.version = '1.0'
+  # `config.assets.precompile` and `config.assets.version` have moved to config/initializers/assets.rb
 
   # Specifies the header that your server uses for sending files.
-  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
-  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
+  # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
-  # Set to :debug to see everything in the log.
-  config.log_level = :info
+  # Use the lowest log level to ensure availability of diagnostic information
+  # when problems arise.
+  config.log_level = :debug
 
   # Prepend all log lines with the following tags.
   # config.log_tags = [ :subdomain, :uuid ]
@@ -55,26 +57,22 @@
   # config.cache_store = :mem_cache_store
 
   # Enable serving of images, stylesheets, and JavaScripts from an asset server.
-  # config.action_controller.asset_host = "http://assets.example.com"
-
-  # Precompile additional assets.
-  # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-  # config.assets.precompile += %w( search.js )
+  # config.action_controller.asset_host = 'http://assets.example.com'
 
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
-  # the I18n.default_locale when a translation can not be found).
+  # the I18n.default_locale when a translation cannot be found).
   config.i18n.fallbacks = true
 
   # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
-  # Disable automatic flushing of the log to improve performance.
-  # config.autoflush_log = false
-
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
 end
diff -Nru -x '*.png' v4.0.1/config/environments/test.rb v4.2.0.rc1/config/environments/test.rb
--- v4.0.1/config/environments/test.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/environments/test.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # The test environment is used exclusively to run your application's
@@ -14,7 +14,7 @@
 
   # Configure static asset server for tests with Cache-Control for performance.
   config.serve_static_assets  = true
-  config.static_cache_control = "public, max-age=3600"
+  config.static_cache_control = 'public, max-age=3600'
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
@@ -31,6 +31,12 @@
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
 
+  # Randomize the order test cases are executed
+  config.active_support.test_order = :random
+
   # Print deprecation notices to the stderr.
   config.active_support.deprecation = :stderr
+
+  # Raises error for missing translations
+  # config.action_view.raise_on_missing_translations = true
 end
diff -Nru -x '*.png' v4.0.1/config/initializers/assets.rb v4.2.0.rc1/config/initializers/assets.rb
--- v4.0.1/config/initializers/assets.rb	1969-12-31 19:00:00.000000000 -0500
+++ v4.2.0.rc1/config/initializers/assets.rb	2014-11-28 23:40:41.000000000 -0500
@@ -0,0 +1,11 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+
+# Add additional assets to the asset load path
+# Rails.application.config.assets.paths << Emoji.images_path
+
+# Precompile additional assets.
+# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+# Rails.application.config.assets.precompile += %w( search.js )
diff -Nru -x '*.png' v4.0.1/config/initializers/cookies_serializer.rb v4.2.0.rc1/config/initializers/cookies_serializer.rb
--- v4.0.1/config/initializers/cookies_serializer.rb	1969-12-31 19:00:00.000000000 -0500
+++ v4.2.0.rc1/config/initializers/cookies_serializer.rb	2014-11-28 23:40:41.000000000 -0500
@@ -0,0 +1,3 @@
+# Be sure to restart your server when you modify this file.
+
+Rails.application.config.action_dispatch.cookies_serializer = :json
diff -Nru -x '*.png' v4.0.1/config/initializers/mime_types.rb v4.2.0.rc1/config/initializers/mime_types.rb
--- v4.0.1/config/initializers/mime_types.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/initializers/mime_types.rb	2014-11-28 23:40:41.000000000 -0500
@@ -2,4 +2,3 @@
 
 # Add new mime types for use in respond_to blocks:
 # Mime::Type.register "text/richtext", :rtf
-# Mime::Type.register_alias "text/html", :iphone
diff -Nru -x '*.png' v4.0.1/config/initializers/secret_token.rb v4.2.0.rc1/config/initializers/secret_token.rb
--- v4.0.1/config/initializers/secret_token.rb	2014-02-21 22:45:12.000000000 -0500
+++ v4.2.0.rc1/config/initializers/secret_token.rb	1969-12-31 19:00:00.000000000 -0500
@@ -1,12 +0,0 @@
-# Be sure to restart your server when you modify this file.
-
-# Your secret key is used for verifying the integrity of signed cookies.
-# If you change this key, all old signed cookies will become invalid!
-
-# Make sure the secret is at least 30 characters and all random,
-# no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
-
-# Make sure your secret_key_base is kept private
-# if you're sharing your code publicly.
-Railsdiff::Application.config.secret_key_base = 'your-secret-token'
diff -Nru -x '*.png' v4.0.1/config/initializers/session_store.rb v4.2.0.rc1/config/initializers/session_store.rb
--- v4.0.1/config/initializers/session_store.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/initializers/session_store.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,3 +1,3 @@
 # Be sure to restart your server when you modify this file.
 
-Railsdiff::Application.config.session_store :cookie_store, key: '_railsdiff_session'
+Rails.application.config.session_store :cookie_store, key: '_railsdiff_session'
diff -Nru -x '*.png' v4.0.1/config/routes.rb v4.2.0.rc1/config/routes.rb
--- v4.0.1/config/routes.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config/routes.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
-Railsdiff::Application.routes.draw do
+Rails.application.routes.draw do
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
diff -Nru -x '*.png' v4.0.1/config/secrets.yml v4.2.0.rc1/config/secrets.yml
--- v4.0.1/config/secrets.yml	1969-12-31 19:00:00.000000000 -0500
+++ v4.2.0.rc1/config/secrets.yml	2014-11-28 23:40:42.000000000 -0500
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
+
+# Your secret key is used for verifying the integrity of signed cookies.
+# If you change this key, all old signed cookies will become invalid!
+
+# Make sure the secret is at least 30 characters and all random,
+# no regular words or you'll be exposed to dictionary attacks.
+# You can use `rake secret` to generate a secure secret key.
+
+# Make sure the secrets in this file are kept private
+# if you're sharing your code publicly.
+
+development:
+  secret_key_base: your-secret-token
+
+test:
+  secret_key_base: your-secret-token
+
+# Do not keep production secrets in the repository,
+# instead read values from the environment.
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
diff -Nru -x '*.png' v4.0.1/config.ru v4.2.0.rc1/config.ru
--- v4.0.1/config.ru	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/config.ru	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
 # This file is used by Rack-based servers to start the application.
 
-require ::File.expand_path('../config/environment',  __FILE__)
+require ::File.expand_path('../config/environment', __FILE__)
 run Rails.application
diff -Nru -x '*.png' v4.0.1/public/404.html v4.2.0.rc1/public/404.html
--- v4.0.1/public/404.html	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/public/404.html	2014-11-28 23:40:41.000000000 -0500
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>The page you were looking for doesn't exist (404)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,9 +57,11 @@
 <body>
   <!-- This file lives in public/404.html -->
   <div class="dialog">
-    <h1>The page you were looking for doesn't exist.</h1>
-    <p>You may have mistyped the address or the page may have moved.</p>
+    <div>
+      <h1>The page you were looking for doesn't exist.</h1>
+      <p>You may have mistyped the address or the page may have moved.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru -x '*.png' v4.0.1/public/422.html v4.2.0.rc1/public/422.html
--- v4.0.1/public/422.html	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/public/422.html	2014-11-28 23:40:41.000000000 -0500
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>The change you wanted was rejected (422)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,9 +57,11 @@
 <body>
   <!-- This file lives in public/422.html -->
   <div class="dialog">
-    <h1>The change you wanted was rejected.</h1>
-    <p>Maybe you tried to change something you didn't have access to.</p>
+    <div>
+      <h1>The change you wanted was rejected.</h1>
+      <p>Maybe you tried to change something you didn't have access to.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru -x '*.png' v4.0.1/public/500.html v4.2.0.rc1/public/500.html
--- v4.0.1/public/500.html	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/public/500.html	2014-11-28 23:40:41.000000000 -0500
@@ -2,17 +2,23 @@
 <html>
 <head>
   <title>We're sorry, but something went wrong (500)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -21,7 +27,8 @@
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
@@ -30,19 +37,19 @@
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
@@ -50,8 +57,10 @@
 <body>
   <!-- This file lives in public/500.html -->
   <div class="dialog">
-    <h1>We're sorry, but something went wrong.</h1>
+    <div>
+      <h1>We're sorry, but something went wrong.</h1>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nru -x '*.png' v4.0.1/public/robots.txt v4.2.0.rc1/public/robots.txt
--- v4.0.1/public/robots.txt	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/public/robots.txt	2014-11-28 23:40:41.000000000 -0500
@@ -1,4 +1,4 @@
-# See http://www.robotstxt.org/wc/norobots.html for documentation on how to use the robots.txt file
+# See http://www.robotstxt.org/robotstxt.html for documentation on how to use the robots.txt file
 #
 # To ban all spiders from the entire site uncomment the next two lines:
 # User-agent: *
diff -Nru -x '*.png' v4.0.1/test/test_helper.rb v4.2.0.rc1/test/test_helper.rb
--- v4.0.1/test/test_helper.rb	2013-11-02 12:27:47.000000000 -0400
+++ v4.2.0.rc1/test/test_helper.rb	2014-11-28 23:40:41.000000000 -0500
@@ -1,14 +1,9 @@
-ENV["RAILS_ENV"] ||= "test"
+ENV['RAILS_ENV'] ||= 'test'
 require File.expand_path('../../config/environment', __FILE__)
 require 'rails/test_help'
 
 class ActiveSupport::TestCase
-  ActiveRecord::Migration.check_pending!
-
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
-  #
-  # Note: You'll currently still have to declare fixtures explicitly in integration tests
-  # -- they do not yet inherit this setting
   fixtures :all
 
   # Add more helper methods to be used by all tests here...
