diff -Nr -U 1000 -x '*.png' v4.0.11/Gemfile v4.1.8/Gemfile
--- v4.0.11/Gemfile	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/Gemfile	2014-11-17 09:12:11.000000000 -0800
@@ -1,45 +1,40 @@
 source 'https://rubygems.org'
 
-# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.0.11'
 
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '4.1.8'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
-
 # Use SCSS for stylesheets
-gem 'sass-rails', '~> 4.0.2'
-
+gem 'sass-rails', '~> 4.0.3'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
-
 # Use CoffeeScript for .js.coffee assets and views
 gem 'coffee-rails', '~> 4.0.0'
-
 # See https://github.com/sstephenson/execjs#readme for more supported runtimes
-# gem 'therubyracer', platforms: :ruby
+# gem 'therubyracer',  platforms: :ruby
 
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
-
 # Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
 gem 'turbolinks'
-
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 1.2'
+gem 'jbuilder', '~> 2.0'
+# bundle exec rake doc:rails generates the API under doc/api.
+gem 'sdoc', '~> 0.4.0',          group: :doc
 
-group :doc do
-  # bundle exec rake doc:rails generates the API under doc/api.
-  gem 'sdoc', require: false
-end
+# Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+gem 'spring',        group: :development
 
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
 # Use unicorn as the app server
 # gem 'unicorn'
 
 # Use Capistrano for deployment
-# gem 'capistrano', group: :development
+# gem 'capistrano-rails', group: :development
 
 # Use debugger
 # gem 'debugger', group: [:development, :test]
+
diff -Nr -U 1000 -x '*.png' v4.0.11/Rakefile v4.1.8/Rakefile
--- v4.0.11/Rakefile	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/Rakefile	2014-11-17 09:12:11.000000000 -0800
@@ -1,6 +1,6 @@
 # Add your own tasks in files placed in lib/tasks ending in .rake,
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
 require File.expand_path('../config/application', __FILE__)
 
-Railsdiff::Application.load_tasks
+Rails.application.load_tasks
diff -Nr -U 1000 -x '*.png' v4.0.11/app/assets/stylesheets/application.css v4.1.8/app/assets/stylesheets/application.css
--- v4.0.11/app/assets/stylesheets/application.css	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/app/assets/stylesheets/application.css	2014-11-17 09:12:11.000000000 -0800
@@ -1,13 +1,15 @@
 /*
  * This is a manifest file that'll be compiled into application.css, which will include all the files
  * listed below.
  *
  * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
  * or vendor/assets/stylesheets of plugins, if any, can be referenced here using a relative path.
  *
- * You're free to add application-wide styles to this file and they'll appear at the top of the
- * compiled file, but it's generally better to create a new file per style scope.
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any styles
+ * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
+ * file per style scope.
  *
- *= require_self
  *= require_tree .
+ *= require_self
  */
diff -Nr -U 1000 -x '*.png' v4.0.11/app/views/layouts/application.html.erb v4.1.8/app/views/layouts/application.html.erb
--- v4.0.11/app/views/layouts/application.html.erb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/app/views/layouts/application.html.erb	2014-11-17 09:12:11.000000000 -0800
@@ -1,14 +1,14 @@
 <!DOCTYPE html>
 <html>
 <head>
   <title>Railsdiff</title>
-  <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
-  <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
 
 <%= yield %>
 
 </body>
 </html>
diff -Nr -U 1000 -x '*.png' v4.0.11/bin/setup v4.1.8/bin/setup
--- v4.0.11/bin/setup	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/bin/setup	1969-12-31 16:00:00.000000000 -0800
@@ -1,29 +0,0 @@
-#!/usr/bin/env ruby
-require 'pathname'
-
-# path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
-
-Dir.chdir APP_ROOT do
-  # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file:
-
-  puts "== Installing dependencies =="
-  system "gem install bundler --conservative"
-  system "bundle check || bundle install"
-
-  # puts "\n== Copying sample files =="
-  # unless File.exist?("config/database.yml")
-  #   system "cp config/database.yml.sample config/database.yml"
-  # end
-
-  puts "\n== Preparing database =="
-  system "bin/rake db:setup"
-
-  puts "\n== Removing old logs and tempfiles =="
-  system "rm -f log/*"
-  system "rm -rf tmp/cache"
-
-  puts "\n== Restarting application server =="
-  system "touch tmp/restart.txt"
-end
diff -Nr -U 1000 -x '*.png' v4.0.11/config/database.yml v4.1.8/config/database.yml
--- v4.0.11/config/database.yml	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/database.yml	2014-11-17 09:12:11.000000000 -0800
@@ -1,25 +1,25 @@
 # SQLite version 3.x
 #   gem install sqlite3
 #
 #   Ensure the SQLite 3 gem is defined in your Gemfile
 #   gem 'sqlite3'
-development:
+#
+default: &default
   adapter: sqlite3
-  database: db/development.sqlite3
   pool: 5
   timeout: 5000
 
+development:
+  <<: *default
+  database: db/development.sqlite3
+
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
+  <<: *default
   database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
 
 production:
-  adapter: sqlite3
+  <<: *default
   database: db/production.sqlite3
-  pool: 5
-  timeout: 5000
diff -Nr -U 1000 -x '*.png' v4.0.11/config/environment.rb v4.1.8/config/environment.rb
--- v4.0.11/config/environment.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/environment.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,5 +1,5 @@
 # Load the Rails application.
 require File.expand_path('../application', __FILE__)
 
 # Initialize the Rails application.
-Railsdiff::Application.initialize!
+Rails.application.initialize!
diff -Nr -U 1000 -x '*.png' v4.0.11/config/environments/development.rb v4.1.8/config/environments/development.rb
--- v4.0.11/config/environments/development.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/environments/development.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,32 +1,37 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # In the development environment your application's code is reloaded on
   # every request. This slows down response time but is perfect for development
   # since you don't have to restart the web server when you make code changes.
   config.cache_classes = false
 
   # Do not eager load code on boot.
   config.eager_load = false
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
 
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
   # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  # Raise an error on page load if there are pending migrations
+  # Raise an error on page load if there are pending migrations.
   config.active_record.migration_error = :page_load
 
   # Debug mode disables concatenation and preprocessing of assets.
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
   config.assets.debug = true
 
+  # Adds additional error checking when serving assets at runtime.
+  # Checks for improperly declared sprockets dependencies.
+  # Raises helpful error messages.
+  config.assets.raise_runtime_errors = true
+
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
 end
diff -Nr -U 1000 -x '*.png' v4.0.11/config/environments/production.rb v4.1.8/config/environments/production.rb
--- v4.0.11/config/environments/production.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/environments/production.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,80 +1,78 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # Code is not reloaded between requests.
   config.cache_classes = true
 
   # Eager load code on boot. This eager loads most of Rails and
-  # your application in memory, allowing both thread web servers
+  # your application in memory, allowing both threaded web servers
   # and those relying on copy on write to perform better.
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
 
   # Full error reports are disabled and caching is turned on.
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
 
   # Enable Rack::Cache to put a simple HTTP cache in front of your application
   # Add `rack-cache` to your Gemfile before enabling this.
   # For large-scale production use, consider using a caching reverse proxy like nginx, varnish or squid.
   # config.action_dispatch.rack_cache = true
 
   # Disable Rails's static asset server (Apache or nginx will already do this).
   config.serve_static_assets = false
 
   # Compress JavaScripts and CSS.
   config.assets.js_compressor = :uglifier
   # config.assets.css_compressor = :sass
 
   # Do not fallback to assets pipeline if a precompiled asset is missed.
   config.assets.compile = false
 
   # Generate digests for assets URLs.
   config.assets.digest = true
 
-  # Version of your assets, change this if you want to expire all your assets.
-  config.assets.version = '1.0'
+  # `config.assets.precompile` and `config.assets.version` have moved to config/initializers/assets.rb
 
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
   # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
   # Set to :debug to see everything in the log.
   config.log_level = :info
 
   # Prepend all log lines with the following tags.
   # config.log_tags = [ :subdomain, :uuid ]
 
   # Use a different logger for distributed setups.
   # config.logger = ActiveSupport::TaggedLogging.new(SyslogLogger.new)
 
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
   # Enable serving of images, stylesheets, and JavaScripts from an asset server.
   # config.action_controller.asset_host = "http://assets.example.com"
 
-  # Precompile additional assets.
-  # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-  # config.assets.precompile += %w( search.js )
-
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
-  # the I18n.default_locale when a translation can not be found).
+  # the I18n.default_locale when a translation cannot be found).
   config.i18n.fallbacks = true
 
   # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
   # Disable automatic flushing of the log to improve performance.
   # config.autoflush_log = false
 
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
 end
diff -Nr -U 1000 -x '*.png' v4.0.11/config/environments/test.rb v4.1.8/config/environments/test.rb
--- v4.0.11/config/environments/test.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/environments/test.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,39 +1,39 @@
-Railsdiff::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # The test environment is used exclusively to run your application's
   # test suite. You never need to work with it otherwise. Remember that
   # your test database is "scratch space" for the test suite and is wiped
   # and recreated between test runs. Don't rely on the data there!
   config.cache_classes = true
 
   # Do not eager load code on boot. This avoids loading your whole application
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
   # Configure static asset server for tests with Cache-Control for performance.
   config.serve_static_assets  = true
-  config.static_cache_control = "public, max-age=3600"
+  config.static_cache_control = 'public, max-age=3600'
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
 
   # Raise exceptions instead of rendering exception templates.
   config.action_dispatch.show_exceptions = false
 
   # Disable request forgery protection in test environment.
   config.action_controller.allow_forgery_protection = false
 
   # Tell Action Mailer not to deliver emails to the real world.
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
 
   # Print deprecation notices to the stderr.
   config.active_support.deprecation = :stderr
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
 end
diff -Nr -U 1000 -x '*.png' v4.0.11/config/initializers/assets.rb v4.1.8/config/initializers/assets.rb
--- v4.0.11/config/initializers/assets.rb	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.8/config/initializers/assets.rb	2014-11-17 09:12:11.000000000 -0800
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+
+# Precompile additional assets.
+# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+# Rails.application.config.assets.precompile += %w( search.js )
diff -Nr -U 1000 -x '*.png' v4.0.11/config/initializers/cookies_serializer.rb v4.1.8/config/initializers/cookies_serializer.rb
--- v4.0.11/config/initializers/cookies_serializer.rb	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.8/config/initializers/cookies_serializer.rb	2014-11-17 09:12:11.000000000 -0800
@@ -0,0 +1,3 @@
+# Be sure to restart your server when you modify this file.
+
+Rails.application.config.action_dispatch.cookies_serializer = :json
\ No newline at end of file
diff -Nr -U 1000 -x '*.png' v4.0.11/config/initializers/mime_types.rb v4.1.8/config/initializers/mime_types.rb
--- v4.0.11/config/initializers/mime_types.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/initializers/mime_types.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,5 +1,4 @@
 # Be sure to restart your server when you modify this file.
 
 # Add new mime types for use in respond_to blocks:
 # Mime::Type.register "text/richtext", :rtf
-# Mime::Type.register_alias "text/html", :iphone
diff -Nr -U 1000 -x '*.png' v4.0.11/config/initializers/secret_token.rb v4.1.8/config/initializers/secret_token.rb
--- v4.0.11/config/initializers/secret_token.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/initializers/secret_token.rb	1969-12-31 16:00:00.000000000 -0800
@@ -1,12 +0,0 @@
-# Be sure to restart your server when you modify this file.
-
-# Your secret key is used for verifying the integrity of signed cookies.
-# If you change this key, all old signed cookies will become invalid!
-
-# Make sure the secret is at least 30 characters and all random,
-# no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
-
-# Make sure your secret_key_base is kept private
-# if you're sharing your code publicly.
-Railsdiff::Application.config.secret_key_base = 'your-secret-token'
diff -Nr -U 1000 -x '*.png' v4.0.11/config/initializers/session_store.rb v4.1.8/config/initializers/session_store.rb
--- v4.0.11/config/initializers/session_store.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/initializers/session_store.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,3 +1,3 @@
 # Be sure to restart your server when you modify this file.
 
-Railsdiff::Application.config.session_store :cookie_store, key: '_railsdiff_session'
+Rails.application.config.session_store :cookie_store, key: '_railsdiff_session'
diff -Nr -U 1000 -x '*.png' v4.0.11/config/routes.rb v4.1.8/config/routes.rb
--- v4.0.11/config/routes.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/config/routes.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,56 +1,56 @@
-Railsdiff::Application.routes.draw do
+Rails.application.routes.draw do
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
   # root 'welcome#index'
 
   # Example of regular route:
   #   get 'products/:id' => 'catalog#view'
 
   # Example of named route that can be invoked with purchase_url(id: product.id)
   #   get 'products/:id/purchase' => 'catalog#purchase', as: :purchase
 
   # Example resource route (maps HTTP verbs to controller actions automatically):
   #   resources :products
 
   # Example resource route with options:
   #   resources :products do
   #     member do
   #       get 'short'
   #       post 'toggle'
   #     end
   #
   #     collection do
   #       get 'sold'
   #     end
   #   end
 
   # Example resource route with sub-resources:
   #   resources :products do
   #     resources :comments, :sales
   #     resource :seller
   #   end
 
   # Example resource route with more complex sub-resources:
   #   resources :products do
   #     resources :comments
   #     resources :sales do
   #       get 'recent', on: :collection
   #     end
   #   end
 
   # Example resource route with concerns:
   #   concern :toggleable do
   #     post 'toggle'
   #   end
   #   resources :posts, concerns: :toggleable
   #   resources :photos, concerns: :toggleable
 
   # Example resource route within a namespace:
   #   namespace :admin do
   #     # Directs /admin/products/* to Admin::ProductsController
   #     # (app/controllers/admin/products_controller.rb)
   #     resources :products
   #   end
 end
diff -Nr -U 1000 -x '*.png' v4.0.11/config/secrets.yml v4.1.8/config/secrets.yml
--- v4.0.11/config/secrets.yml	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.8/config/secrets.yml	2014-11-17 09:12:11.000000000 -0800
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
+
+# Your secret key is used for verifying the integrity of signed cookies.
+# If you change this key, all old signed cookies will become invalid!
+
+# Make sure the secret is at least 30 characters and all random,
+# no regular words or you'll be exposed to dictionary attacks.
+# You can use `rake secret` to generate a secure secret key.
+
+# Make sure the secrets in this file are kept private
+# if you're sharing your code publicly.
+
+development:
+  secret_key_base: your-secret-token
+
+test:
+  secret_key_base: your-secret-token
+
+# Do not keep production secrets in the repository,
+# instead read values from the environment.
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
diff -Nr -U 1000 -x '*.png' v4.0.11/public/404.html v4.1.8/public/404.html
--- v4.0.11/public/404.html	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/public/404.html	2014-11-17 09:12:11.000000000 -0800
@@ -1,58 +1,67 @@
 <!DOCTYPE html>
 <html>
 <head>
   <title>The page you were looking for doesn't exist (404)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
     border-bottom-color: #BBB;
     border-top: #B00100 solid 4px;
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
 
 <body>
   <!-- This file lives in public/404.html -->
   <div class="dialog">
-    <h1>The page you were looking for doesn't exist.</h1>
-    <p>You may have mistyped the address or the page may have moved.</p>
+    <div>
+      <h1>The page you were looking for doesn't exist.</h1>
+      <p>You may have mistyped the address or the page may have moved.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nr -U 1000 -x '*.png' v4.0.11/public/422.html v4.1.8/public/422.html
--- v4.0.11/public/422.html	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/public/422.html	2014-11-17 09:12:11.000000000 -0800
@@ -1,58 +1,67 @@
 <!DOCTYPE html>
 <html>
 <head>
   <title>The change you wanted was rejected (422)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
     border-bottom-color: #BBB;
     border-top: #B00100 solid 4px;
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
 
 <body>
   <!-- This file lives in public/422.html -->
   <div class="dialog">
-    <h1>The change you wanted was rejected.</h1>
-    <p>Maybe you tried to change something you didn't have access to.</p>
+    <div>
+      <h1>The change you wanted was rejected.</h1>
+      <p>Maybe you tried to change something you didn't have access to.</p>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nr -U 1000 -x '*.png' v4.0.11/public/500.html v4.1.8/public/500.html
--- v4.0.11/public/500.html	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/public/500.html	2014-11-17 09:12:11.000000000 -0800
@@ -1,57 +1,66 @@
 <!DOCTYPE html>
 <html>
 <head>
   <title>We're sorry, but something went wrong (500)</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
   body {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
     font-family: arial, sans-serif;
+    margin: 0;
   }
 
   div.dialog {
-    width: 25em;
-    margin: 4em auto 0 auto;
+    width: 95%;
+    max-width: 33em;
+    margin: 4em auto 0;
+  }
+
+  div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
     border-bottom-color: #BBB;
     border-top: #B00100 solid 4px;
     border-top-left-radius: 9px;
     border-top-right-radius: 9px;
     background-color: white;
-    padding: 7px 4em 0 4em;
+    padding: 7px 12% 0;
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
   h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  body > p {
-    width: 33em;
-    margin: 0 auto 1em;
-    padding: 1em 0;
+  div.dialog > p {
+    margin: 0 0 1em;
+    padding: 1em;
     background-color: #F7F7F7;
     border: 1px solid #CCC;
     border-right-color: #999;
+    border-left-color: #999;
     border-bottom-color: #999;
     border-bottom-left-radius: 4px;
     border-bottom-right-radius: 4px;
     border-top-color: #DADADA;
     color: #666;
-    box-shadow:0 3px 8px rgba(50, 50, 50, 0.17);
+    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
   </style>
 </head>
 
 <body>
   <!-- This file lives in public/500.html -->
   <div class="dialog">
-    <h1>We're sorry, but something went wrong.</h1>
+    <div>
+      <h1>We're sorry, but something went wrong.</h1>
+    </div>
+    <p>If you are the application owner check the logs for more information.</p>
   </div>
-  <p>If you are the application owner check the logs for more information.</p>
 </body>
 </html>
diff -Nr -U 1000 -x '*.png' v4.0.11/test/test_helper.rb v4.1.8/test/test_helper.rb
--- v4.0.11/test/test_helper.rb	2014-10-31 00:45:52.000000000 -0700
+++ v4.1.8/test/test_helper.rb	2014-11-17 09:12:11.000000000 -0800
@@ -1,15 +1,10 @@
-ENV["RAILS_ENV"] ||= "test"
+ENV['RAILS_ENV'] ||= 'test'
 require File.expand_path('../../config/environment', __FILE__)
 require 'rails/test_help'
 
 class ActiveSupport::TestCase
-  ActiveRecord::Migration.check_pending!
-
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
-  #
-  # Note: You'll currently still have to declare fixtures explicitly in integration tests
-  # -- they do not yet inherit this setting
   fixtures :all
 
   # Add more helper methods to be used by all tests here...
 end
