diff -Nru v4.1.1/Gemfile v4.1.2.rc3/Gemfile
--- v4.1.1/Gemfile	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc3/Gemfile	2014-06-23 10:39:24.000000000 -0700
@@ -2,7 +2,7 @@
 
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.1.1'
+gem 'rails', '4.1.2.rc3'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
 # Use SCSS for stylesheets
diff -Nru v4.1.1/config/environments/production.rb v4.1.2.rc3/config/environments/production.rb
--- v4.1.1/config/environments/production.rb	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc3/config/environments/production.rb	2014-06-23 10:39:24.000000000 -0700
@@ -32,8 +32,7 @@
   # Generate digests for assets URLs.
   config.assets.digest = true
 
-  # Version of your assets, change this if you want to expire all your assets.
-  config.assets.version = '1.0'
+  # `config.assets.precompile` has moved to config/initializers/assets.rb
 
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
diff -Nru v4.1.1/config/initializers/assets.rb v4.1.2.rc3/config/initializers/assets.rb
--- v4.1.1/config/initializers/assets.rb	1969-12-31 16:00:00.000000000 -0800
+++ v4.1.2.rc3/config/initializers/assets.rb	2014-06-23 10:39:24.000000000 -0700
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+
+# Precompile additional assets.
+# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+# Rails.application.config.assets.precompile += %w( search.js )
diff -Nru v4.1.1/test/test_helper.rb v4.1.2.rc3/test/test_helper.rb
--- v4.1.1/test/test_helper.rb	2014-05-06 11:06:14.000000000 -0700
+++ v4.1.2.rc3/test/test_helper.rb	2014-06-23 10:39:24.000000000 -0700
@@ -4,9 +4,6 @@
 
 class ActiveSupport::TestCase
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
-  #
-  # Note: You'll currently still have to declare fixtures explicitly in integration tests
-  # -- they do not yet inherit this setting
   fixtures :all
 
   # Add more helper methods to be used by all tests here...
